### DISCUSSION TITLE
What is upstream DNS when using unbound?

### DISCUSSION THREAD

**Discussion by quyleanh** [complexorganizations/wireguard-manager#216] [Q&A]
As I understanding, when using unbound, I will connect directly to NS from `root.hints` file, right?
So it will be more privacy, prevent dns leak? And there is no upstream DNS server like `8.8.8.8` and `4.4.4.4`?
Thank you.

**Comment by ghost** [comment]
https://ipleak.net/

Check it here.

During installation we check if there is a nameserver in `/etc/resolve.conf` and if we find one we don't do anything and continue but if there are no resolvers than we add one, but this is replaced once u complete the installation.

**Comment by quyleanh** [comment]
What do you mean "we add one"?
My old `/etc/resolv.conf` is
```
nameserver 127.0.0.53
options edns0 trust-ad
```
And the new one after install this script is
```
nameserver 127.0.0.1
nameserver ::1
```

Btw, I check 2 peer, the DNS servers are different. So maybe my understand is correct?

**Reply by ghost** [reply]
Some people don't have a dns server in their config file and then the script wont work properly, if you have a dns server than nothing will change.

**Reply by quyleanh** [reply]
Understand now.
So my current config is fine, right?

**Reply by ghost** [reply]
Yes

**Reply by quyleanh** [reply]
Thank you.

**Reply by ghost** [reply]
> Thank you.

Hello,
Star, Sponsors, Fork, Follow is not mandatory but is much appreciated.
- Prajwal

**Reply by quyleanh** [reply]
Already did. Thank you.
